lib/erlef/data/query/academic_paper.ex

```diff
@@ -0,0 +1,121 @@
+defmodule Erlef.Data.Query.AcademicPaper do
+  @moduledoc """
+  Module for the academic paper queries
+  """
+  import Ecto.Query, only: [from: 2]
+
+  alias Erlef.Data.Schema.AcademicPaper
+  alias Erlef.Data.Repo
+
+  @doc """
+  Returns all active academic papers. Used for admin functionality
+  """
+  @spec all(repo :: Ecto.Repo.t()) :: [AcademicPaper.t()]
+  def all(repo \\ Repo) do
+    from(p in AcademicPaper,
+      where: is_nil(p.deleted_at),
+      order_by: [p.original_publish_date, p.title]
+    )
+    |> repo.all()
+  end
+
+  @doc """
+  Returns a academic paper by id. Used for admin functionality
+  """
+  @spec get(id :: binary(), repo :: Ecto.Repo.t()) :: AcademicPaper.t() | nil
+  def get(id, repo \\ Repo) do
+    from(p in AcademicPaper,
+      where: is_nil(p.deleted_at),
+      where: p.id == ^id
+    )
+    |> repo.one()
+  end
+
+  @doc """
+  Returns all active published academic papers
+  """
+  @spec published(repo :: Ecto.Repo.t()) :: [AcademicPaper.t()]
+  def published(repo \\ Repo) do
+    from(p in AcademicPaper,
+      where: is_nil(p.deleted_at),
+      where: not is_nil(p.published_at),
+      order_by: [p.original_publish_date, p.title]
+    )
+    |> repo.all()
+  end
+
+  @doc """
+  Returns all active unpublished academic papers, Used for admin functionality
+  """
+  @spec unpublished(repo :: Ecto.Repo.t()) :: [AcademicPaper.t()]
+  def unpublished(repo \\ Repo) do
+    from(p in AcademicPaper,
+      where: is_nil(p.deleted_at),
+      where: is_nil(p.published_at),
+      order_by: [p.original_publish_date, p.title]
+    )
+    |> repo.all()
+  end
+
+  @doc """
+  Creates a new academic paper. Used for admin functionality
+  """
+  @spec create(params :: map(), repo :: Ecto.Repo.t()) ::
+          {:ok, AcademicPaper.t()} | {:error, Ecto.Changeset.t()}
+  def create(params, repo \\ Repo) do
+    %AcademicPaper{}
+    |> AcademicPaper.changeset(params)
+    |> case do
```

Comment:
Non-blocking comment. IMO this is a bad use of pipes. I'm not worried about fixing it up right now though.

Reply:
Ok, I'll bite on this one. Curious why? Can you provide a reason or is this just an opinion?

Reply:
At the end of the day this compiles to the same erlang, so there is no technical foundation in my argument in that regard, that's why I listed this as non-blocking.  However, I do not believe pipes were ever intended to be used in this way, it just so happens they can be. My arguments would be that this creates cognitive overload and is generally hard to read, but that is subjective as I do know some people fancy this style and obviously if they do, it is not cognitive overload for them and not hard to read for them. 

 My feeling on this have been quite loud and known for some time though: https://github.com/rrrene/credo-proposals/issues/37

If we are trying to carve out consistency, then I need to make myself heard on these issues, OTOH, I'm not concerned in that things just need to work right now, but even still I'd rather flag this as "I do not agree, we'll come back to this".

Reply:
Cool. I see where you are coming from. It is definitely a "style" preference. To me it's easier to follow but like you said, it's subjective :)